apps/api/src/app/environments-v1/usecases/output-renderers/hydrate-email-schema.usecase.ts

```diff
@@ -117,10 +117,11 @@ export class HydrateEmailSchemaUseCase {
     node,
     placeholderAggregation: PlaceholderAggregation
   ) {
-    const resolvedValue = this.getValueByPath(masterPayload, node.attrs.id);
     const { fallback } = node.attrs;
+    const variableName = node.attrs.id;
+    const buildLiquidJSDefault = (mailyFallback: string) => (mailyFallback ? ` | default: '${mailyFallback}'` : '');
```

Comment:
Consider moving all the variable calculations from start to end into the `buildLiquidJSVariable` function defined once either on the class or as a single util. That is combine line 122 and 133.

Reply:
Let's also add an extra step of sanitization in the above function that caters for weird cases where the `variableName` ends with ` | `.

Reply:
Lastly, this transformation should happena cross all steps, not just email.

Reply:
> Consider moving all the variable calculations from start to end into the buildLiquidJSVariable function defined once either on the class or as a single util. That is combine line 122 and 133.

If you refer to the `show` resolver, i totally agree, this was postponed until we have full show support.


> Let's also add an extra step of sanitization in the above function that caters for weird cases where the variableName ends with |.

I love the idea but i believe we can postpone such sanitization because liquid js does not care about filter suffixes.
![image](https://github.com/user-attachments/assets/da6d853b-23b8-4b02-8f44-634bd7f6e7b8)


> Lastly, this transformation should happena cross all steps, not just email.

can you elaborate please what do you mean by this transformation